workflows/build-base-on-changes.py

```diff
@@ -0,0 +1,415 @@
+
+        # On 20 and 22 these packages are built by the snap and not pulled from
+        # the archive.
+        built_by_snap = ['console-conf', 'probert-common',
+                         'probert-network', 'subiquitycore']
+        # Look out for changes. We could return on first change, but we'll
+        # print all changes for the moment for debugging purposes.
+        for pkg in dpkg['packages']:
+            [pkgName, pkgVersion] = pkg.split('=')
+            # pkgName can have a :<arch> suffix, like :amd64 or :i386. The few
+            # i386 packages that are in the manifest are also present as amd64
+            # packages, so we do not worry with filtering.
+            pkgName = pkgName.split(':')[0]
+            if pkgName in built_by_snap and (
+                    core_version == 20 or core_version == 22):
+                continue
+            if pkgName not in snap2version:
+                print('unexpected error, package {} from {} '
+                      'not found in the archive'.format(pkgName, base))
+                sys.exit(1)
+            if apt_pkg.version_compare(pkgVersion, snap2version[pkgName]) < 0:
+                print('change in {}: {} package version updated ({} -> {})'.
+                      format(base, pkgName, pkgVersion, snap2version[pkgName]))
+                changed = True
+
+    return changed
+
+
+# We know that the branch has changed if there are no date tags in HEAD. If we
+# find one, branch changes are already in some published snap.
+def check_branch_changed():
+    gtag = subprocess.run(['git', 'tag', '--points-at', 'HEAD'],
+                          check=True, stdout=subprocess.PIPE)
+    # We expect <date><month><day> with an optional dash followed by a number:
+    # 20250510, 20263001, 20250510-1 and 20263001-2 are valid.
+    date_re = re.compile(r'[0-9]{8}(-[0-9]+)?')
+    for tag in gtag.stdout.splitlines():
+        tag = tag.decode("utf-8")
+        print('found tag', tag)
+        if re.match(date_re, tag):
+            return False
+
+    print('no date tag found, triggering build')
+    return True
+
+
+# Get tag that we will use in the build.
+def get_build_tag():
+    today = datetime.today().strftime('%Y%m%d')
+    gtag = subprocess.run(['git', 'tag', '--points-at', 'HEAD'],
+                          check=True, stdout=subprocess.PIPE)
+    # We expect <today> followed by an optional dash and sequence number.
+    date_re = re.compile(r'{}(-[0-9]+)?'.format(today))
+    last_seq = 0
+    today_found = False
+    for tag in gtag.stdout.splitlines():
+        tag = tag.decode("utf-8")
+        m = re.match(date_re, tag)
+        if m:
+            today_found = True
+            if m.group(1) != '':
+                # Remove the dash and get sequence number
+                seq = int(m.group(1)[1:])
+                if seq > last_seq:
+                    last_seq = seq
+
+    if today_found:
+        return today + '-' + str(last_seq + 1)
+
+    return today
```

Comment:
I wonder if to simplify, we could just always use a sequence number and start with 1.

Reply:
The intention is that this will be almost always a date as I prefer to avoid the sequence in the base versions. But I have introduced sequence as a safeguard just in case we occasionally need more than one daily build, which I hope won't happen often.